{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "The sql extension is already loaded. To reload it, use:\n",
      "  %reload_ext sql\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "'Connected: anonymous@homo_sapiens_core_92_38'"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "%load_ext sql\n",
    "%config SqlMagic.autocommit=False\n",
    "%sql mysql+pymysql://anonymous@ensembldb.ensembl.org/homo_sapiens_core_92_38\n",
    "            "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      " * mysql+pymysql://anonymous@ensembldb.ensembl.org/homo_sapiens_core_92_38\n",
      "73 rows affected.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       "    <tr>\n",
       "        <th>Tables_in_homo_sapiens_core_92_38</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "        <td>alt_allele</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "        <td>alt_allele_attrib</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "        <td>alt_allele_group</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "        <td>analysis</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "        <td>analysis_description</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "        <td>assembly</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "        <td>assembly_exception</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "        <td>associated_group</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "        <td>associated_xref</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "        <td>attrib_type</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "        <td>coord_system</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "        <td>data_file</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "        <td>density_feature</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "        <td>density_type</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "        <td>dependent_xref</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "        <td>ditag</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "        <td>ditag_feature</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "        <td>dna</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "        <td>dna_align_feature</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "        <td>dna_align_feature_attrib</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "        <td>exon</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "        <td>exon_transcript</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "        <td>external_db</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "        <td>external_synonym</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "        <td>gene</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "        <td>gene_archive</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "        <td>gene_attrib</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "        <td>genome_statistics</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "        <td>identity_xref</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "        <td>interpro</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "        <td>intron_supporting_evidence</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "        <td>karyotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "        <td>map</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "        <td>mapping_session</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "        <td>mapping_set</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "        <td>marker</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "        <td>marker_feature</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "        <td>marker_map_location</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "        <td>marker_synonym</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "        <td>meta</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "        <td>meta_coord</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "        <td>misc_attrib</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "        <td>misc_feature</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "        <td>misc_feature_misc_set</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "        <td>misc_set</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "        <td>object_xref</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "        <td>ontology_xref</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "        <td>operon</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "        <td>operon_transcript</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "        <td>operon_transcript_gene</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "        <td>peptide_archive</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "        <td>prediction_exon</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "        <td>prediction_transcript</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "        <td>protein_align_feature</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "        <td>protein_feature</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "        <td>repeat_consensus</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "        <td>repeat_feature</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "        <td>seq_region</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "        <td>seq_region_attrib</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "        <td>seq_region_mapping</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "        <td>seq_region_synonym</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "        <td>simple_feature</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "        <td>stable_id_event</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "        <td>supporting_feature</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "        <td>transcript</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "        <td>transcript_attrib</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "        <td>transcript_intron_supporting_evidence</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "        <td>transcript_supporting_feature</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "        <td>translation</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "        <td>translation_attrib</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "        <td>unmapped_object</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "        <td>unmapped_reason</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "        <td>xref</td>\n",
       "    </tr>\n",
       "</table>"
      ],
      "text/plain": [
       "[('alt_allele',),\n",
       " ('alt_allele_attrib',),\n",
       " ('alt_allele_group',),\n",
       " ('analysis',),\n",
       " ('analysis_description',),\n",
       " ('assembly',),\n",
       " ('assembly_exception',),\n",
       " ('associated_group',),\n",
       " ('associated_xref',),\n",
       " ('attrib_type',),\n",
       " ('coord_system',),\n",
       " ('data_file',),\n",
       " ('density_feature',),\n",
       " ('density_type',),\n",
       " ('dependent_xref',),\n",
       " ('ditag',),\n",
       " ('ditag_feature',),\n",
       " ('dna',),\n",
       " ('dna_align_feature',),\n",
       " ('dna_align_feature_attrib',),\n",
       " ('exon',),\n",
       " ('exon_transcript',),\n",
       " ('external_db',),\n",
       " ('external_synonym',),\n",
       " ('gene',),\n",
       " ('gene_archive',),\n",
       " ('gene_attrib',),\n",
       " ('genome_statistics',),\n",
       " ('identity_xref',),\n",
       " ('interpro',),\n",
       " ('intron_supporting_evidence',),\n",
       " ('karyotype',),\n",
       " ('map',),\n",
       " ('mapping_session',),\n",
       " ('mapping_set',),\n",
       " ('marker',),\n",
       " ('marker_feature',),\n",
       " ('marker_map_location',),\n",
       " ('marker_synonym',),\n",
       " ('meta',),\n",
       " ('meta_coord',),\n",
       " ('misc_attrib',),\n",
       " ('misc_feature',),\n",
       " ('misc_feature_misc_set',),\n",
       " ('misc_set',),\n",
       " ('object_xref',),\n",
       " ('ontology_xref',),\n",
       " ('operon',),\n",
       " ('operon_transcript',),\n",
       " ('operon_transcript_gene',),\n",
       " ('peptide_archive',),\n",
       " ('prediction_exon',),\n",
       " ('prediction_transcript',),\n",
       " ('protein_align_feature',),\n",
       " ('protein_feature',),\n",
       " ('repeat_consensus',),\n",
       " ('repeat_feature',),\n",
       " ('seq_region',),\n",
       " ('seq_region_attrib',),\n",
       " ('seq_region_mapping',),\n",
       " ('seq_region_synonym',),\n",
       " ('simple_feature',),\n",
       " ('stable_id_event',),\n",
       " ('supporting_feature',),\n",
       " ('transcript',),\n",
       " ('transcript_attrib',),\n",
       " ('transcript_intron_supporting_evidence',),\n",
       " ('transcript_supporting_feature',),\n",
       " ('translation',),\n",
       " ('translation_attrib',),\n",
       " ('unmapped_object',),\n",
       " ('unmapped_reason',),\n",
       " ('xref',)]"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "%sql show tables;\n"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
